cpp/include/cudf/column/column.hpp

```diff
@@ -75,6 +76,21 @@ class column {
    */
   column(column&& other) noexcept;
 
+  /**
+   * @brief Move the contents from `other` to create a new column.
+   *
+   * After the move, `other.data() == NULL`
+   *
+   * @param other The device_uvector whose contents will be moved into the new column.
+   */
+  template <typename T>
+  column(rmm::device_uvector<T>&& other) noexcept
+    : _type{cudf::type_to_id<T>()},
+      _size{static_cast<cudf::size_type>(other.size())},
+      _data{other.release()}
```

Comment:
This needs to be constrained in some way to prevent construction from a `device_uvector<string_view>` or `device_uvector<list_view>`.

It should probably be an `enable_if` for `is_fixed_width<T>()`.

Reply:
We probably want the same condition as here: `cudf::is_numeric<T>() or cudf::is_chrono<T>()`

https://github.com/rapidsai/cudf/blob/e7e5f456f4d7a47c76fb17cd20ca3ea037e15618/cpp/include/cudf/column/column_view.hpp#L428

Reply:
Another reason this should probably just be a `make_fixed_width_column` factory function instead of a constructor.
This factory: https://docs.rapids.ai/api/libcudf/stable/group__column__factories.html#ga595aed5d1e6f3d4b3d8da9eeffed916d
takes a buffer for a null mask. It seems reasonable to add one that accepts a device-uvector.